### PR TITLE
Update SwiftSyntax (2023-01-16)

### PR DESCRIPTION
**Pull request by jpsim**
Probably can't merge until a new release of SwiftSyntax is cut to avoid the SwiftPM dependency resolution issues, but there have been a lot of breaking changes in the weeks since the last release that I didn't want to get too far behind.

I'm also hoping that the memory corruption fix in https://github.com/apple/swift-syntax/pull/1190 helps with the tsan crashes I've been seeing occasionally on CI.

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/Idiomatic/UnavailableFunctionRule.swift [hunk 1]
```diff
@@ -123,13 +123,24 @@ private extension AttributeListSyntax? {
...
-                arg.entry.as(TokenSyntax.self)?.tokenKind == .contextualKeyword("unavailable")
+            let attributeName = attr.attributeName.as(SimpleTypeIdentifierSyntax.self)?.name.text
+            return attributeName == "available" && arguments.contains { arg in
+                arg.entry.as(TokenSyntax.self)?.tokenKind.isUnavailable == true
             }
         }
     }
 }
 
+private extension TokenKind {
```
Comment (SimplyDanny) [inline Source/SwiftLintFramework/Rules/Idiomatic/UnavailableFunctionRule.swift:134]: Move to `SwiftLint+SwiftSyntax.swift`?

File: Source/SwiftLintFramework/Rules/Performance/ContainsOverFilterCountRule.swift [hunk 2]
```diff
@@ -60,10 +60,8 @@ private extension ContainsOverFilterCountRule {
 
 private extension TokenKind {
```
Comment (SimplyDanny) [inline Source/SwiftLintFramework/Rules/Performance/ContainsOverFilterCountRule.swift:61]: Move to `SwiftSyntax+SwiftLint.swift`?

File: Source/SwiftLintFramework/Rules/Idiomatic/ConvenienceTypeRule.swift [hunk 3]
```diff
@@ -222,9 +222,20 @@ private extension AttributeListSyntax? {
                 return false
             }
 
-            return attr.attributeName.tokenKind == .contextualKeyword("available") && arguments.contains { arg in
-                arg.entry.as(TokenSyntax.self)?.tokenKind == .contextualKeyword("unavailable")
+            let attributeName = attr.attributeName.as(SimpleTypeIdentifierSyntax.self)?.name.text
```
[the remark this answers is not in the document]
Comment (jpsim) [reply Source/SwiftLintFramework/Rules/Idiomatic/ConvenienceTypeRule.swift:225]: Definitely.

File: Source/SwiftLintFramework/Rules/Idiomatic/UnavailableFunctionRule.swift (hunk 1 above)
Comment (jpsim) [reply Source/SwiftLintFramework/Rules/Idiomatic/UnavailableFunctionRule.swift:134]: Good idea.

File: Source/SwiftLintFramework/Rules/Performance/ContainsOverFilterCountRule.swift (hunk 2 above)
Comment (jpsim) [reply Source/SwiftLintFramework/Rules/Performance/ContainsOverFilterCountRule.swift:61]: Good idea.

**Comment by jpsim** [comment]
Linting the Brave & Wordpress projects causes SwiftLint to crash with this PR (`SIGTRAP`). I'll look into that.

**Comment by jpsim** [comment]
All other OSSCheck reports appear to fix false positives.

**Pull request closed without merging**